beekeeper-scheduler/src/main/java/com/expediagroup/beekeeper/scheduler/service/ExpiredHousekeepingMetadataSchedulerService.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.scheduler.service;
+
+import static java.lang.String.format;
+
+import static com.expediagroup.beekeeper.core.model.LifecycleEventType.EXPIRED;
+
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.expediagroup.beekeeper.core.error.BeekeeperException;
+import com.expediagroup.beekeeper.core.model.HousekeepingEntity;
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.core.model.HousekeepingStatus;
+import com.expediagroup.beekeeper.core.model.LifecycleEventType;
+import com.expediagroup.beekeeper.core.monitoring.TimedTaggable;
+import com.expediagroup.beekeeper.core.repository.HousekeepingMetadataRepository;
+
+@Service
+public class ExpiredHousekeepingMetadataSchedulerService implements SchedulerService {
+
+  private final LifecycleEventType LIFECYCLE_EVENT_TYPE = EXPIRED;
+  private final HousekeepingMetadataRepository housekeepingMetadataRepository;
+
+  @Autowired
+  public ExpiredHousekeepingMetadataSchedulerService(HousekeepingMetadataRepository housekeepingMetadataRepository) {
+    this.housekeepingMetadataRepository = housekeepingMetadataRepository;
+  }
+
+  @Override
+  public LifecycleEventType getLifecycleEventType() {
+    return LIFECYCLE_EVENT_TYPE;
+  }
+
+  @Override
+  @TimedTaggable("metadata-scheduled")
+  public void scheduleForHousekeeping(HousekeepingEntity housekeepingEntity) {
+    HousekeepingMetadata housekeepingTable = createOrUpdateHousekeepingTable(
+        (HousekeepingMetadata) housekeepingEntity);
+    try {
+      housekeepingMetadataRepository.save(housekeepingTable);
```

Comment:
maybe it would be good to log something here with some basic information about what's being scheduled. the same goes for the unreferenced handler. currently nothing is logged which is a little annoying.

Reply:
Lets leave this for a different PR :)